server/app/views/FileUploadViewStrategy.java

```diff
@@ -152,6 +152,26 @@ protected Tag renderUploadButton(Params params) {
         .withId(FILEUPLOAD_SUBMIT_FORM_ID);
   }
 
+  protected Tag renderPreviousButton(Params params) {
+    int previousBlockIndex = params.blockIndex() - 1;
+    String redirectUrl;
+
+    if (previousBlockIndex >= 0) {
+      redirectUrl =
+          routes.ApplicantProgramBlocksController.previous(
+                  params.applicantId(), params.programId(), previousBlockIndex, params.inReview())
+              .url();
+    } else {
+      redirectUrl =
+          routes.ApplicantProgramReviewController.preview(params.applicantId(), params.programId())
+              .url();
+    }
+    return a().attr(HREF, redirectUrl)
+        .withText(params.messages().at(MessageKey.BUTTON_PREVIOUS_SCREEN.getKeyName()))
+        .withClasses(ApplicantStyles.BUTTON_BLOCK_PREVIOUS)
+        .withId("cf-block-previous");
+  }
+
```

Comment:
suggestion (blocking): looks like this code was copied without modification from https://github.com/seattle-uat/civiform/blob/main/server/app/views/applicant/ApplicantProgramBlockEditView.java#L174-L193

Rather than introducing code duplication, please move the original method somewhere it can be shared. One option for this could be to create a class in this same package `class ApplicantBaseView extends BaseHtmlView` that implements `Tag renderPreviousButton(Params params)`, and then have other view classes in this package inherit from it.

For that to work you'll likely need to move the definition of views.applicant.ApplicantProgramBlockEditView.Params to that shared base class as well.